File: app/src/main/java/com/stho/software/nyota/utilities/Clock.java
package com.stho.software.nyota.utilities;

public class Clock {
    private long startMillis;
    private final long intervalInMillis;

    public Clock(long intervalInMillis) {
        this.intervalInMillis = intervalInMillis;
        this.startMillis = 0;
    }

    public boolean timeOver() {
        long millis = System.currentTimeMillis();
        if (millis - startMillis > intervalInMillis) {
            startMillis = millis;
            return true;
        }
        else {
            return false;
        }
    }
}

File: app/src/main/java/com/stho/nyota/sky/utilities/Clock.kt
package com.stho.nyota.sky.utilities

class Clock(private val intervalInMillis: Long) {
    private var startMillis: Long = 0
    fun timeOver(): Boolean {
        val millis = System.currentTimeMillis()
        return if (millis - startMillis > intervalInMillis) {
            startMillis = millis
            true
        } else {
            false
        }
    }

}